### PR TITLE
add separate tasks for OAuth clients without user

### PR DESCRIPTION
**Pull request by melvinsoft**
This extra task, allow us the flexibility of creating the Client with and without a username attached.

### CODE REVIEW

File: playbooks/roles/oauth_client_setup/tasks/main.yml [hunk 1]
```diff
@@ -37,5 +37,24 @@
...
+    {{ COMMON_BIN_DIR }}/python.edxapp {{ COMMON_BIN_DIR }}/manage.edxapp lms --settings={{ COMMON_EDXAPP_SETTINGS }}
+    create_oauth2_client
+    {{ item.url_root }}
+    "{{ item.url_root }}/complete/edx-oidc/"
+    confidential
+    --client_name {{ item.name }}
+    --client_id {{ item.id }}
+    --client_secret {{ item.secret }}
+    --trusted
+    --logout_uri {{ item.logout_uri | default("") }}
```
Comment (OmarIthawi) [inline playbooks/roles/oauth_client_setup/tasks/main.yml]: @melvinsoft I think we should keep it in the same list and use  

```suggestion
    ...
    --logout_uri {{ item.logout_uri | default("") }}
    {% if item.username %} --username {{ item.username }} {% endif %}
    ...
```

File: playbooks/roles/oauth_client_setup/tasks/main.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@
 #
 #
 
-- name: create OAuth2 Clients
+- name: create OAuth2 Clients with user
```
Comment (OmarIthawi) [inline playbooks/roles/oauth_client_setup/tasks/main.yml]:
```suggestion
- name: create OAuth2 Clients
```

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
Thanks @melvinsoft! This works for now, but I hope that we can simplify it back into a single step.

File: playbooks/roles/oauth_client_setup/tasks/main.yml (hunk 1 above)
Comment (OmarIthawi) [reply playbooks/roles/oauth_client_setup/tasks/main.yml]: See https://gist.github.com/halberom/794c06598f40ccc31560#file-opt2_play-yml

**Comment by melvinsoft** [comment]
@OmarIthawi Please take another look, I follow your lead and came up with a better solution than before, keeping it simple.

File: playbooks/roles/oauth_client_setup/tasks/main.yml [hunk 3]
```diff
@@ -33,7 +33,7 @@
     --client_secret {{ item.secret }}
     --trusted
     --logout_uri {{ item.logout_uri | default("") }}
-    --username {{ item.username }}
+    {% if item.username is defined %} --username {{ item.username }} {% endif %}
```
Comment (OmarIthawi) [inline playbooks/roles/oauth_client_setup/tasks/main.yml:36]: Is the `is defined` super needed? can't we just use `{% if item.username %}`

**Review by OmarIthawi — APPROVED** [review]
LGTM. Thanks @melvinsoft! 

I have one question. Feel free to merge regardless.

File: playbooks/roles/oauth_client_setup/tasks/main.yml (hunk 3 above)
Comment (melvinsoft) [reply playbooks/roles/oauth_client_setup/tasks/main.yml:36]: No, I tried that first, and I was getting an error, "The username key doesn't exists" or something like that. We need to explicitly check if it's defined.

**Pull request merged**